diff --git a/setup.py b/setup.py
index 0b465ac..199ea3e 100644
--- a/setup.py
+++ b/setup.py
@@ -506,12 +506,6 @@ elif (sys.platform[0:3] == 'win'):
 else:
     ext_modules = _ext_modules(ibm_db_include, library, ibm_db_lib, ibm_db_lib_runtime)
         
-if('win32' not in sys.platform):
-    if sys.version_info >= (3, ):        
-        if (sys.platform != 'zos'):
-           _checkGcc()
-           _checkPythonHeaderFile()
-
 #'Operating System :: z/OS', pypi upload fails with error - Not a valid classifier
 setup( name    = PACKAGE,
     version = VERSION,
